det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+
+    Args:
+        cloud_dict: (dict) A dictionary with the timestamp as key and point cloud as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and transformed point cloud as value
+    """
+    for k, v in cloud_dict.items():
+        SE3 = get_city_SE3_egovehicle_at_sensor_t(int(k), dataset, log_id)
+        cloud_dict[k] = SE3.transform_point_cloud(v)
+    return cloud_dict
+
+
+def grids_group_and_SE3(cloud_dict: Dict, dataset: str, log_id: str, n_t: int) -> Dict:
+    """Groups five point clouds and performs the necessary SE3 transformation
+
+    Args:
+        cloud_dict: (dict) A dictionary with the timestamp as key and transformed point cloud as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        n_t: (int) Number of timestamps per group
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and five consecutive point clouds as value
+    """
+    timestamps = []
+    for key in sorted(cloud_dict.keys()):
+        timestamps.append(key)
+        if len(timestamps) == n_t:
+            t0_to_map_SE3 = get_city_SE3_egovehicle_at_sensor_t(
+                int(timestamps[0]), dataset, log_id
+            )
+            map_to_t0_SE3 = t0_to_map_SE3.inverse()
+            rotation = np.eye(3)
+            translation = np.array([XSize / 2, YSize / 2, 0])
+            t0_to_occ_SE3 = se3.SE3(rotation, translation)
+            grids = []
+            for ts in timestamps:
+                pc = cloud_dict[ts]
+                pc = map_to_t0_SE3.transform_point_cloud(pc)
+                pc = t0_to_occ_SE3.transform_point_cloud(pc)
+                grids.append(pc)
+            cloud_dict[timestamps[0]] = grids
+            timestamps = timestamps[1:]
+    for ts in timestamps:
+        del cloud_dict[ts]
+    return cloud_dict
+
+
+def load_all_boxes(dataset: str, log_id: str, timestamp_list: List[int]) -> Dict:
+    """Loads all bounding boxes from the Argoverse Tracking set
+
+    Args:
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        timestamp_list: (list[int]) List of timestamps to load point clouds from
+    Returns:
+        data_dict: (dict) A dictionary with the timestamp as key and all bounding boxes as value
+
+    Note: This bounding box representation can be found at
+        https://github.com/argoai/argoverse-api/blob/master/argoverse/data_loading/object_label_record.py#L232
+    """
+    file_path = os.path.join(dataset, log_id, "per_sweep_annotations_amodal")
+    data_dict = {}
+    for timestamp in timestamp_list:
+        with open(
+                os.path.join(file_path, f"tracked_object_labels_{timestamp}.json")
+        ) as jsonfile:
```

Comment:
i would prefer to use our library function from argoverse-api if possible, instead of adding the json.loads calls here:
```python
from argoverse.utils.json_utils import read_json_file
arr = read_json_file( os.path.join(file_path, f"tracked_object_labels_{timestamp}.json") )
```

https://github.com/argoai/argoverse-api/blob/master/argoverse/utils/json_utils.py#L9